### PR TITLE
2 user intefaces

### PR DESCRIPTION
**Pull request by ec105**
Combing user intefaces to main branch
completed structuing : 
UIinsidechatroom
UIinsidechatroom_backend
UIoutsidechatroom
UIoutsidechatroom_backend
Loginpage
Loginusecase
user

### CODE REVIEW

**Review by SamandarVijay — APPROVED** [review]

File: src/main/java/tutorial/User.java [hunk 1]
```diff
@@ -15,11 +22,14 @@ public class User {
      * @param id the User's id as assigned when the User is created
      */
 
-    public User(String displayName, int id) {
```
Comment (is1006) [inline src/main/java/tutorial/User.java:18]: copy of next line, needs to be deleted

File: src/main/java/tutorial/User.java [hunk 2]
```diff
@@ -15,11 +22,14 @@ public class User {
...
+    public User(String displayName, int id, String username, String password) {
         this.displayName = displayName;
-        this.id = id; //need to determine how we will be creating id and ensuring it does not already exist //
```
Comment (is1006) [inline src/main/java/tutorial/User.java:20]: can remove this line and the line below, we will be getting rid of id as a variable entirely

**Review by is1006 — CHANGES_REQUESTED** [review]

**Review by is1006 — APPROVED** [review]

**Review by SamandarVijay — DISMISSED** [review]

**Review by hu-jeff — COMMENTED** [review]
Why are we gathering the username and password from the command line?

**Comment by ec105** [comment]
> Why are we gathering the username and password from the command line?

it is needed for loginsystem, when we init the user object, we assumed it is created when username and password and them both should be saved

**Comment by hu-jeff** [comment]
> > Why are we gathering the username and password from the command line?
> 
> it is needed for loginsystem, when we init the user object, we assumed it is created when username and password and them both should be saved

But why are we getting this information from the command line? We should be getting it from the UI. I think we need parameters that call that function that are the username and password.

**Comment by ec105** [comment]
> > > Why are we gathering the username and password from the command line?
> > 
> > 
> > it is needed for loginsystem, when we init the user object, we assumed it is created when username and password and them both should be saved
> 
> But why are we getting this information from the command line? We should be getting it from the UI. I think we need parameters that call that function that are the username and password.

the info is received from the UI, but just need to be stored in User class when initializing

**Comment by hu-jeff** [comment]
> > > > Why are we gathering the username and password from the command line?
> > > 
> > > 
> > > it is needed for loginsystem, when we init the user object, we assumed it is created when username and password and them both should be saved
> > 
> > 
> > But why are we getting this information from the command line? We should be getting it from the UI. I think we need parameters that call that function that are the username and password.
> 
> the info is received from the UI, but just need to be stored in User class when initializing

The beginlogin function loops infinitely until there is a response in the console.

**Review by SamandarVijay — DISMISSED** [review]

**Review by is1006 — DISMISSED** [review]

**Review by hu-jeff — CHANGES_REQUESTED** [review]
we shouldn't read from system.in

**Review by hu-jeff — CHANGES_REQUESTED** [review]
Sam's portion should be to submit the login.

In the LoginPage, line 31. Why are we using a loop for the login system? It seems no matter what the loop only runs once. Furthermore, to check if a login is valid, we should be using the accounts storage interface to check the username and password. The input for the username and password should be from the user as well. I think the checking of usernames and password should be a method that the user interface calls.

Furthermore, InputMismatchException appears to only be thrown by a Scanner. In this case, I can't see where the Scanner is being used.

In the LoginPage method, secondChoice, we should not be using system.out or system.in. 

In the file LoginUseCase, I don't think we need an array list of all users. Our Storage methods don't have any method which can give all users. Instead, what we should do is use the account storage interface to check if a username and password are valid and then we can get the user data from the user storage interface. We should similarly do the same for creating a new user.

**Comment by SamandarVijay** [comment]
> Sam's portion should be to submit the login.
> 
> In the LoginPage, line 31. Why are we using a loop for the login system? It seems no matter what the loop only runs once. Furthermore, to check if a login is valid, we should be using the accounts storage interface to check the username and password. The input for the username and password should be from the user as well. I think the checking of usernames and password should be a method that the user interface calls.
> 
> Furthermore, InputMismatchException appears to only be thrown by a Scanner. In this case, I can't see where the Scanner is being used.
> 
> In the LoginPage method, secondChoice, we should not be using system.out or system.in.
> 
> In the file LoginUseCase, I don't think we need an array list of all users. Our Storage methods don't have any method which can give all users. Instead, what we should do is use the account storage interface to check if a username and password are valid and then we can get the user data from the user storage interface. We should similarly do the same for creating a new user.

I did it!

**Pull request closed without merging**